frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -588,9 +604,29 @@ pub mod pallet {
         /// However, caller have to have deposit amount.
         /// TODO: weight
         #[pallet::weight(T::WeightInfo::payout_stakers_alive_staked(T::MaxStakings::get()))]
-        pub fn register(origin: OriginFor<T>, contract_id: T::AccountId) -> DispatchResult {
-            // TODO: impls
-            Ok(())
+        pub fn register(
+            origin: OriginFor<T>,
+            contract_id: SmartContract<T::AccountId>,
+        ) -> DispatchResultWithPostInfo {
+            let operator = ensure_signed(origin)?;
+            let mut ledger = Self::ledger(&operator).ok_or(Error::<T>::NotController)?;
+            ensure!(
+                !RegisteredDapps::<T>::contains_key(&operator),
+                Error::<T>::AlreadyRegisteredContract
+            );
```

Comment:
@Dinonard the check is good, just the Error is wrong. If developer already registered a contract, this ensure will fire error. I Changed this error name to AlreadyUsedDeveloperAccount

Reply:
But what is missing is the check if contract is already registered by someone else.
I will add the check and test